instrumentation/src/main/java/io/opentelemetry/android/internal/services/AppInfoService.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.internal.services;
+
+import android.content.Context;
+import android.os.Build;
+import android.os.storage.StorageManager;
+import androidx.annotation.RequiresApi;
+import androidx.annotation.WorkerThread;
+import java.io.File;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Utility to get information about the host app.
+ *
+ * <p>This class is internal and not for public use. Its APIs are unstable and can change at any
+ * time.
+ */
+public class AppInfoService implements Service {
+    private final Context appContext;
+    private final Logger logger = Logger.getLogger("AppInfoService");
+
+    public AppInfoService(Context appContext) {
+        this.appContext = appContext;
+    }
+
+    public File getCacheDir() {
+        return appContext.getCacheDir();
+    }
+
+    @WorkerThread
+    public long getAvailableCacheSpace(long maxSpaceNeeded) {
```

Comment:
It can be frustrating to troubleshoot a problem and then later discover that a seemingly harmless "getter" method has side effects (like allocating cache memory here). I think a rename like `ensureCacheSpaceAvailable(long)` helps with this -- that way the caller is more aware. 

A javadoc is also warranted. Right now the reader has no clue what the return value means.

Reply:
Thanks, it's updated now.